src/Sarif/Readers/LineMappingStreamReader.cs

```diff
@@ -41,15 +41,55 @@ internal class LineMappingStreamReader : StreamReader
         // Track and attempt to correct for it.
         private readonly OverflowCorrector _overflowCorrector;
 
-        public LineMappingStreamReader(Stream stream) : base(stream)
+        public LineMappingStreamReader(Stream stream) : base(FindBomWidth(stream, out int bomWidth))
         {
+            // Record the width of any BOM, so that byte offsets we return are correct (base StreamReader will read and hide BOM)
+            _bytesReadPreviously = bomWidth;
+
             // (1, 1) is the 0th byte, so the line number before the read is 1 and there is one character (the newline before the first line) before the first read.
             _firstLineNumber = 1;
             _firstLineCharsBeforeBuffer = 1;
 
             _overflowCorrector = new OverflowCorrector();
         }
 
+        private static Stream FindBomWidth(Stream stream, out int bomWidth)
```

Comment:
>private [](start = 8, length = 7)

Should have unit tests (using our `testCase` pattern).

Reply:
Added. I wasn't able to get tests to run, so I wasn't excited about adding them at that point.